final_assignment/alex-mitan/conf.py

```diff
@@ -0,0 +1,161 @@
+                    build_log.write("        INSTALL COMMAND: " + command + '\n')
+
+                    # run a script
+                    if command == "run_script":
+                        scr_command = cmd_dict["run_script"]["command"]
+                        # change directory
+                        if cmd_dict["run_script"]["cwd"]:
+                            if not os.system("cd " + cmd_dict["run_script"]["cwd"]):
+                                build_log.write("Changed directory to: " + cmd_dict["run_script"]["cwd"] + '\n')
+                            else:
+                                build_log.write("ERROR: changing directory failed!" + '\n')
+                                return "error"
+
+                        # set env. variables
+                        for var in cmd_dict["run_script"]["env_variables"]:
+                            var_value = cmd_dict["run_script"]["env_variables"][var]
+                            build_log.write("Setting variable: " + var + "to" + str(var_value) + '\n')
+                            os.environ[var] = var_value
+                            if os.environ[var] == var_value:
+                                build_log.write("+   Successfully set " + var + "to" + var_value + '\n')
+
+                        # execute command
+                        scr_exit_code = os.system(scr_command)
+                        build_log.write("script exit command: " + str(scr_exit_code) + '\n')
+                        build_log.write("script attempt: " + "1" + '\n')
+
+                        # if we have to repeat it
+                        if cmd_dict["run_script"]["check_exit_code"]:
+                            if cmd_dict["run_script"]["retry_interval"]:
+                                scr_interv = cmd_dict["run_script"]["retry_interval"]
+                            attempts = 1
+                            while attempts <= cmd_dict["run_script"]["attempts"] and scr_exit_code:
+                                # sleep for interval
+                                if scr_interv:
+                                    time.sleep(scr_interv)
+                                build_log.write("Checking exit code and retrying" + '\n')
+                                attempts += 1
+                                build_log.write("script attempt: " + attempts + '\n')
+                                scr_exit_code = os.system(scr_command)
+                        if cmd_dict["run_script"]["shell"]:
+                            pass
+                            # TODO: shell stuff
+
+                # AFTER INSTALL:
+                if phase_list == "after_install":
+                    build_log.write("        INSTALL COMMAND: " + command)
+                    if command == "reboot":
+                        if cmd_dict["reboot"]["method"] == "soft":
+                            build_log.write("Performing soft reboot")
+                            if work_os == "linux":
```

Comment:
Check-urile astea de os atat de dese nu sunt foarte indicate. Puteai sa faci in asa fel incat sa verifici platforma o singura data la inceputul programului.